code/__HELPERS/icons.dm

```diff
@@ -48,4 +85,20 @@ proc/adjust_brightness(var/color, var/value)
 	RGB[1] = Clamp(RGB[1]+value,0,255)
 	RGB[2] = Clamp(RGB[2]+value,0,255)
 	RGB[3] = Clamp(RGB[3]+value,0,255)
-	return rgb(RGB[1],RGB[2],RGB[3])
\ No newline at end of file
+	return rgb(RGB[1],RGB[2],RGB[3])
+
+/proc/AverageColor(var/icon/I)

```

Comment:
I was really surprised this wasn't in the graphics library to start with.
At least it might be useful in the future.